tests/TaskTest.php

```diff
@@ -170,181 +161,151 @@ public function testGetAllActions() {
-                );
+        $actualStatusesFromMap = array_keys($actualMapStatusActions);
 
-            $oneFilteredExpectedStatus =
-                current($oneFilteredExpectedStatusConstant);
+        sort($expectedStatuses);
+        sort($actualStatusesFromMap);
 
-            $actionsByStatusStr = array_keys($statusInfo['actions']);
+        $this->assertEquals(
+            $expectedStatuses,
+            $actualStatusesFromMap,
+            'WRONG INITIAL KEYS IN MAP STATUS ACTIONS'
+        );
 
-            if (count($actionsByStatusStr) === 0) {
-                array_push(
-                    $dataSet,
-                    [$oneFilteredExpectedStatus, null]
-                );
-                continue;
-            }
+    }
 
-            $actionsByStatus = array_reduce(
-                $actionsByStatusStr,
+    public function testStructureMapActionStatus() {
 
-                function ($curry, $oneActionByStatusStr) use ($allExpectedActionConstants) {
-                    $oneActionArr = array_filter(
+        $actualMapActionStatus = $this->taskInst->getMapActionStatus();
+        $this->assertIsArray(
+            $actualMapActionStatus,
+            'MAP ACTION STATUS MUST BE ARRAY'
+        );
 
-                        $allExpectedActionConstants,
-                        function ($oneAction) use ($oneActionByStatusStr) {
-                            return $oneAction['EN'] === $oneActionByStatusStr;
-                        }
+        foreach ($actualMapActionStatus as $status) {
+            $this->assertIsString(
+                $status,
+                'STATUS MUST BE STRING'
+            );
+        }
 
-                    );
-                    array_push($curry, current($oneActionArr));
-                    return $curry;
-                },
+        $expectedActions = array_values(
+            self::getClassConstants(Task::class)[self::PREFIX_ACTION]
+        );
 
-                array()
-            );
 
-            array_push(
-                $dataSet,
-                [$oneFilteredExpectedStatus, $actionsByStatus]
-            );
+        $actualActionsFromMap = array_keys($actualMapActionStatus);
 
-        };
 
-        return $dataSet;
-    }
+        sort($expectedActions);
+        sort($actualActionsFromMap);
 
-    /**
-     * @dataProvider dataActionsForStatus
-     */
-    public function testGetActionsByStatus($status, $expectedActions) {
-        $actualActions = $this->taskInst->getActionsByStatus($status);
         $this->assertEquals(
             $expectedActions,
-            $actualActions,
-            "WRONG ACTIONS FOR STATUS - {$status['EN']}"
-        );
-    }
-
-    /*
-TODO -
-проверить , что по заданному ACTION возвращается верный STATUS
-*/
-    public function testGetStatusAfterAction() {
-        $this->markTestIncomplete(
-            'AWAITING IMPLEMENTATION'
+            $actualActionsFromMap,
+            'WRONG INITIAL KEYS IN MAP ACTION STATUS'
         );
     }
 
```

Comment:
public function testStructureMapActionStatus()
    {
       $this->assertEquals(
            $this->taskInst->getMapActionStatus(),
            [
                Task::ACTION_CANCEL   => Task::STATUS_CANCELED,
                Task::ACTION_RESPOND  => Task::STATUS_WORKING,
                Task::ACTION_COMPLETE => Task::STATUS_DONE,
                Task::ACTION_REJECT   => Task::STATUS_FAILED
            ],
            'WRONG INITIAL KEYS IN MAP ACTION STATUS'
        );
    }